src/datanode/src/datanode.rs

```diff
@@ -493,6 +473,63 @@ impl DatanodeBuilder {
         Ok(engines)
     }
 
+    /// Builds [MitoEngine] according to options.
+    async fn build_mito_engine(
+        opts: &DatanodeOptions,
+        object_store_manager: ObjectStoreManagerRef,
+        config: MitoConfig,
+    ) -> Result<MitoEngine> {
+        // TODO(niebayes): validate wal config, for e.g. providing a remote wal for standalone mode is not allowed.
```

Comment:
Looks like it's allowed, so we must modify the comment. But I think we should do the other validations right now, don't make it to be a `TODO`.

Reply:
The TODO is removed since we allow standalone mode to also use remote wal.
The type system already applies enough constraints on the wal config and currently there's no need to do extra validations.